reader.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2022 by Posit Software, PBC
```

Comment:
2022 can be updated to 2023 if we wanted to.

Reply:
Done.